### PR TITLE
organize docs into showcase, docs, and examples

### PR DESCRIPTION
**Pull request by Joshuafolorunsho**
- Docs can house every important concept we want the users to know
image

- This should house all of the primitive APIs
image

- WIP for showcase. This should showcase all of our examples.
image

### CODE REVIEW

File: docs/next.config.mjs [hunk 1]
```diff
@@ -7,6 +7,5 @@ const withNextra = nextra({
 });
 
 export default withNextra({
-  basePath: '/docs',
```
Comment (adeleke5140) [inline docs/next.config.mjs:10]: we have this `basePath` for the reverse proxy on the website domain.

Comment (adeleke5140) [reply docs/next.config.mjs:10]: we could take it out but we'd have to verify that it still works when you visit `mastra.ai/docs`

Comment (Joshuafolorunsho) [reply docs/next.config.mjs:10]: There is a /docs folder now, so we don't need this anymore.

**Review by adeleke5140 — APPROVED** [review]
lgtm

**Pull request merged**